event_sink_clickhouse/management/commands/dump_data_to_clickhouse.py

```diff
@@ -64,15 +65,12 @@ def dump_target_objects_to_clickhouse(
             log.info(
                 f"{sink.model} {index}: Submitting {object_id} for dump to ClickHouse, reason '{reason}'."
             )
+            objects_to_submit.append(object_id)
 
-            dump_data_to_clickhouse.apply_async(
-                kwargs={
-                    "sink_module": sink.__module__,
-                    "sink_name": sink.__class__.__name__,
-                    "object_id": str(object_id),
-                    "connection_overrides": connection_overrides,
-                }
-            )
+            if len(objects_to_submit) % batch_size == 0:
+                sink.dump(objects_to_submit, many=True)
+                submitted_objects.extend(objects_to_submit)
+                objects_to_submit = []
```

Comment:
You'll need to do another `dump` if there's anything in `objects_to_submit` at the end of the loop since there may not be an even batch_size number of total objects.

Reply:
thanks, corrected